homeassistant/components/flume/sensor.py

```diff
@@ -0,0 +1,94 @@
+"""Sensor for displaying the number of result from Flume."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from pyflume import FlumeDeviceList, FlumeData
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Flume Sensor"
+
+CONF_CLIENT_ID = "client_id"
+CONF_CLIENT_SECRET = "client_secret"
+FLUME_TYPE_SENSOR = 2
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Flume sensor."""
+    _username = config[CONF_USERNAME]
+    _password = config[CONF_PASSWORD]
+    _client_id = config[CONF_CLIENT_ID]
+    _client_secret = config[CONF_CLIENT_SECRET]
+    time_zone = str(hass.config.time_zone)
+    name = config[CONF_NAME]
+
+    flume_devices = FlumeDeviceList(_username, _password, _client_id, _client_secret)
+
+    try:
+        for device in flume_devices.device_list:
+            if device["type"] == FLUME_TYPE_SENSOR:
+                flume = FlumeData(
```

Comment:
By creating a new object for each entity, you will also hit the server with a new API call on each update for each entity. So 10 entities is 10 API calls. This object should be shared and only 1 update should be made .

Reply:
I believe I implemented throttle which will help address this. The user_id and device_id is within the URI so I am unable to do one call for multiple devices.

Reply:
I'm not certain if this is resolved so I will leave it open until I receive confirmation from you.